### PR TITLE
Double Round Rolls

### PR DESCRIPTION
**Pull request by Geevies**
* Players who have played a role the previous round will now be the last in line for candidates chosen for the same role in the current round. You only count as having played a role if you join it before one hour has elapsed.

Feedback thread: https://forums.aurorastation.org/topic/16572-feedback-double-round-rolls/

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by FayeLovesU** [comment]
Long overdue IMO. This is a great and I'd almost wager necessary change for people who play few-slot roles that are highly contested like Investigator.

**Comment by Jaraci** [comment]
Does this apply to latejoiners or only people who take the role at roundstart? Just curious.

**Comment by Geevies** [comment]
> 
> 
> Does this apply to latejoiners or only people who take the role at roundstart? Just curious.

just pushed an update. it counts for roundstart and latejoiners, but latejoining after one hour has passed will not cause it

**Comment by NonQueueingMatt** [comment]
I personally disagree with latejoiners counting. I don't think it should apply to people who queued for a slot and lost the roll and joined later. At the same time, I don't think it should apply to people who join into a slot that wasn't contested.

**Comment by AnselmKonrad** [comment]
I dont agree with making this applicable to late joining. but how does this stop someone from exactly stopping someone from playing the role again for the next round? does this just entirely stop someone from selecting that role at all? does it just put them at the end of a queue? do they get a decreased chance instead? I can see this only killing the server pop for people who main a single character which is, frankly, a lot of people.

**Comment by Geevies** [comment]
> 
> 
> I personally disagree with latejoiners counting. I don't think it should apply to people who queued for a slot and lost the roll and joined later. At the same time, I don't think it should apply to people who join into a slot that wasn't contested.

why?



> 
> 
> I dont agree with making this applicable to late joining. I can understand perhaps somehow reducing it but, the choosing of people getting a role is supposed to be random already. Mainly though, how does this stop someone from exactly stopping someone from playing the role again for the next round? does this just entirely stop someone from selecting that role at all? does it just put them at the end of a queue? do they get a decreased chance instead? I can see this only killing the server pop for people who main a single character which is, frankly, a lot of people.

last in queue

**Comment by AnselmKonrad** [comment]
> Last in queue

So this stops them from joining next round if anyone contests the role at all?

**Comment by Geevies** [comment]
> 
> 
> > Last in queue
> 
> So this stops them from joining next round if anyone contests the role at all?

yeah. but it checks high, medium, and low too

File: code/controllers/subsystems/job.dm [hunk 1]
```diff
@@ -113,6 +113,8 @@
 			position_limit = job.get_spawn_positions()
 		if((job.current_positions < position_limit) || position_limit == -1)
 			Debug("Player: [player] is now Rank: [rank], JCP:[job.current_positions], JPL:[position_limit]")
```
Comment (Arrow768) [inline code/controllers/subsystems/job.dm:115]: The time played should be a configuration value.
It should also be possible to disable this system via a special configuration value. i.e. -1

**Review by Arrow768 — CHANGES_REQUESTED** [review]

**Review by Arrow768 — CHANGES_REQUESTED** [review]
It seems like I never actually submitted that review, so here we go

The default should maintain the existing behavior.

**Review by Arrow768 — APPROVED** [review]

**Review by JohnWildkins — APPROVED** [review]

**Comment by NonQueueingMatt** [comment]
Conflicts. Will get re-discussed if anyone re-opens this.

**Pull request closed without merging**